cli/cmd/tap.go

```diff
@@ -109,30 +107,26 @@ func requestTapFromApi(client pb.ApiClient, targetName string, resourceType stri
 			Pod: targetName,
 		}
 	default:
-		return "", fmt.Errorf("unsupported resourceType [%s]", resourceType)
+		return fmt.Errorf("unsupported resourceType [%s]", resourceType)
 	}
 
 	rsp, err := client.Tap(context.Background(), req)
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	return renderTap(rsp)
+	return renderTap(w, rsp)
 }
 
-func renderTap(rsp pb.Api_TapClient) (string, error) {
-	var buffer bytes.Buffer
-	w := tabwriter.NewWriter(&buffer, 0, 0, 0, ' ', tabwriter.AlignRight)
-	err := writeTapEvenToBuffer(rsp, w)
+func renderTap(w io.Writer, rsp pb.Api_TapClient) error {
+	tableWriter := tabwriter.NewWriter(w, 0, 0, 0, ' ', tabwriter.AlignRight)
+	err := writeTapEvenToBuffer(rsp, tableWriter)
 	if err != nil {
-		return "", err
+		return err
 	}
-	w.Flush()
+	tableWriter.Flush()
```

Comment:
how about:
```
defer tableWriter.Flush()
return writeTapEvenToBuffer(...
```

Reply:
Won't this make it harder to understand that something happens then something else happens? i.e. first write, then flush

Reply:
`Flush()` is more like a "close", in that it's only called once when you're all done. i'm pretty used to seeing this pattern:
```
thing = Create()
defer thing.Close()
...
thing.Stuff()
```
...i also thought saving a couple lines by removing the error check could make it slightly clearer.

again this is all cosmetic, i'm ok with whatever you think makes more sense.

Reply:
👍  to @siggy's comment here. I think it makes sense to `defer tableWriter.Flush()` immediately after tableWriter is instantiated.